addons/html_builder/static/src/website_builder/website_mass_mailing/mailing_list_subscribe_option_plugin.js

```diff
@@ -0,0 +1,119 @@
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+import { Plugin } from "@html_editor/plugin";
+import { registry } from "@web/core/registry";
+import { user } from "@web/core/user";
+import { _t } from "@web/core/l10n/translation";
+import { NewsletterSubscribeCommonOption } from "./newsletter_subscribe_common_option";
+import { getSelectorParams } from "@html_builder/utils/utils";
+import { applyFunDependOnSelectorAndExclude } from "@html_builder/plugins/utils";
+
+class MailingListSubscribeOptionPlugin extends Plugin {
+    static id = "mailingListSubscribeOption";
+    static dependencies = ["remove", "savePlugin"];
+    static shared = ["fetchMailingLists"];
+    resources = {
+        builder_actions: [
+            {
+                toggleThanksMessage: {
+                    apply: ({ editingElement }) => {
+                        this.setThanksMessageVisibility(editingElement, true);
+                    },
+                    clean: ({ editingElement }) => {
+                        this.setThanksMessageVisibility(editingElement, false);
+                    },
+                    isApplied: ({ editingElement }) =>
+                        editingElement
+                            .querySelector(".js_subscribed_wrap")
+                            .classList.contains("o_enable_preview"),
+                },
+            },
+        ],
+        on_add_element_handlers: this.onAddElement.bind(this),
+    };
+
+    setup() {
+        this.mailingListSubscribeOptionSelectorParams = getSelectorParams(
+            this.getResource("builder_options"),
+            NewsletterSubscribeCommonOption
+        );
+    }
+
+    setThanksMessageVisibility(editingElement, isVisible) {
+        const toSubscribeEl = editingElement.querySelector(".js_subscribe_wrap");
+        const thanksMessageEl = editingElement.querySelector(".js_subscribed_wrap");
+        thanksMessageEl.classList.toggle("o_enable_preview", isVisible);
+        thanksMessageEl.classList.toggle("o_disable_preview", !isVisible);
+        toSubscribeEl.classList.toggle("o_enable_preview", !isVisible);
+        toSubscribeEl.classList.toggle("o_disable_preview", isVisible);
+    }
+
+    async onAddElement({ elementToAdd }) {
+        for (const mailingListSubscribeOptionSelector of this
+            .mailingListSubscribeOptionSelectorParams) {
+            applyFunDependOnSelectorAndExclude(
```

Comment:
Maybe a good time to introduce the `applyTo` concept in the `applyFunDependOnSelectorAndExclude` utils ? But if you prefer, you can add a `TODO` and do it in another PR.